### PR TITLE
Rails generator for resources

### PR DESCRIPTION
**Pull request by kevintraver**
Initial attempt at creating a Rails generator for JSONAPI resource files

### CODE REVIEW

File: lib/generators/jsonapi/templates/jsonapi_resource.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<% module_namespacing do -%>
+class <%= class_name %>Resource > JSONAPI::Resource
```
Comment (lgebhardt) [inline lib/generators/jsonapi/templates/jsonapi_resource.rb]: Should use <

``` ruby
class <%= class_name %>Resource < JSONAPI::Resource
```

**Comment by lgebhardt** [comment]
Cool, thanks. Looks like a good start. This is now creating a resources directory in the project. We should add this to the .gitignore file or clean up the files after the tests.

Also it seems the resources are replaced if the generator is run again. I don't use the Rails generators much so I don't know if this is standard. But it seems risky as you could overwrite some of your work if it accidentally run it again (which I'm sure I would do at times with a missed keystroke in selecting command history).

I don't know if generators usually go this far, but it would be cool to interrogate the model and setup the default relationships and attributes. That's certainly not needed at first, but it seems useful.

**Comment by kevintraver** [comment]
Thanks for the feedback. If you are inheriting from the standard rails generator it looks like get a confirmation to overwrite.

I agree it would be cool to have more integration. I think the next step would be to add parameters for attributes.

**Comment by lgebhardt** [comment]
I just tested it out in a real project and it does warn you if the file exists. So no issue there. However it created the resource directory in the root of the project, and by convention the directory should be `app/resources`.

**Comment by barelyknown** [comment]
Nice idea @kevintraver. I don't use generators all that often, but I think that this is a friendly feature for new users of `jsonapi-resources`.

File: lib/generators/jsonapi/USAGE [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Description:
+    Generator for JSONAPI Resources
+
+Example:
+    rails generate jsonapi:resource Post
+
+    This will create:
+        resources/post_resource.rb
```
Comment (lgebhardt) [inline lib/generators/jsonapi/USAGE]: app/resources/post_resource.rb

File: test/lib/generators/jsonapi/resource_generator_test.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'test_helper'
+require 'generators/jsonapi/resource_generator'
+
+module Jsonapi
+  class ResourceGeneratorTest < Rails::Generators::TestCase
+    tests ResourceGenerator
+    destination Rails.root.join('../resources')
```
Comment (lgebhardt) [inline test/lib/generators/jsonapi/resource_generator_test.rb:7]: Could you put the generated files into a temp directory. Maybe `../test/temp/generators`. And then we can add that to .gitignore.

Comment (kevintraver) [inline test/lib/generators/jsonapi/resource_generator_test.rb:7]: Im sure theres a proper way to teardown, I just need to find out how to do it.

**Comment by kevintraver** [comment]
I've fixed the teardown/cleanup issue, and improved the tests.

**Comment by lgebhardt** [comment]
Thanks!

**Comment by jondkinney** [comment]
@kevintraver @lgebhardt any plans to add this generator into the Rails config.generators hooks so that it happens automatically (if configured to do so) when generating a rails resource? `rails g resource user` for example, would also generate a jsonapi resource.

**Pull request merged**